File: composeApp/src/commonMain/kotlin/felix/livinglink/ui/settings/SettingsScreen.kt
package felix.livinglink.ui.settings

import SettingsScreenLocalizables
import androidx.compose.runtime.Composable
import felix.livinglink.ui.common.BackAwareScaffold
import felix.livinglink.ui.common.state.LoadableStatefulView

@Composable
fun SettingsScreen(viewModel: SettingsViewModel) {
    BackAwareScaffold(
        navigator = viewModel.navigator,
        title = SettingsScreenLocalizables.navigationTitle(),
    ) { innerPadding ->
        LoadableStatefulView(
            viewModel = viewModel,
            modifier = innerPadding,
            content = { loadableData, data ->
                SettingsScreenContent(
                    loadableData = loadableData,
                    data = data,
                    viewModel = viewModel
                )
            }
        )
    }
}

File: iosApp/iosApp/SettingsScreen/SettingsScreen.swift
//
//  SettingsScreen.swift
//  iosApp
//
//  Created by Felix Fischer on 26.03.25.
//  Copyright © 2025 orgName. All rights reserved.
//

import ComposeApp
import SwiftUI

struct SettingsScreen: View {
    let viewModel: SettingsViewModel
    let localizables = SettingsScreenLocalizables()

    var body: some View {
        LoadableStatefulView(
            viewModel: viewModel,
            buildAlert: { (error: SettingsScreenError) in
                error.asAlert(
                    navigator: viewModel.navigator,
                    dismiss: viewModel.closeError
                )
            },
            content: { loadableData, data in
                SettingsContentScreen(
                    loadableData: loadableData,
                    data: data,
                    viewModel: viewModel
                )
            }
        )
        .navigationTitle(localizables.navigationTitle.localized)
    }
}

private typealias SettingsScreenError = LoadableViewModelStateCombinedError<KotlinNothing, KotlinNothing, NetworkError>
